crates/floresta-watch-only/src/descriptors.rs

```diff
@@ -0,0 +1,1073 @@
+//! Holds the methods and types to help handling descriptors exactly how floresta needs.
+//!
+//! The watch-only wallet doesnt store any private keys so [`DescriptorUnit`] holds a [`Descriptor<DescriptorPublicKey>`] which should satisfy
+//! all use cases that floresta might need, [`DescriptorUnit`] also holds some metadata such as a `DerivationRange` that accepts operations to extend and sum indices.
+//!
+//! Everything in this module works around [`DescriptorUnit`] and to work with them. Heres a short description about each struct and what you should do with them.
+//!
+//! [`DescriptorRequest`] - The input scheme for the `importdescriptors` RPC that helps us to batch produce [`DescriptorUnit`]s, it is pretty extensive in options and should support
+//! any kind of descriptor as an input, besides the invalid ones since the validation for correctnessonly happens in [`DescriptorRequest::into_unit`], since this should be the only
```

Comment:
```suggestion
//! any kind of descriptor as an input, besides the invalid ones since the validation for correctness only happens in [`DescriptorRequest::into_unit`], since this should be the only
```